### ISSUE TITLE
[BUG] VM is stuck in terminating after a reboot 

### ISSUE THREAD

**Issue by masterdubs** [harvester/harvester#994]
**Describe the bug**
The VM is stuck in terminating after a reboot or sometimes arbitrary. It happen on the master node ( first k3s node installed) not on the second one.
I need to delete the pod manually ( using force ) to get the vm recreated.

**Environment:**
 - Harvester ISO version:  0.2.0
 - Baremetal 2 servers, SSD

``` Name:                      virt-launcher-79bec1b1-12c6-49e5-8e8e-b0ae4469e4af-8k6b7
Namespace:                 default
Priority:                  0
Node:                      node1/10.0.0.38
Start Time:                Sun, 27 Jun 2021 18:07:37 +0400
Labels:                    harvesterhci.io/creator=harvester
                           harvesterhci.io/vmName=79bec1b1-12c6-49e5-8e8e-b0ae4469e4af
                           kubevirt.io=virt-launcher
                           kubevirt.io/created-by=f0da20ff-5cca-4c1e-a454-484447f1e4f7
Annotations:               harvesterhci.io/diskNames: ["79bec1b1-12c6-49e5-8e8e-b0ae4469e4af-rootdisk-1z8mz"]
                           harvesterhci.io/sshNames: [""]
                           k8s.v1.cni.cncf.io/network-status:
                             [{
                                 "name": "",
                                 "interface": "eth0",
                                 "ips": [
                                     "10.52.0.230"
                                 ],
                                 "mac": "1a:7c:3c:8e:bc:5c",
                                 "default": true,
                                 "dns": {}
                             },{
                                 "name": "default/public",
                                 "interface": "net1",
                                 "mac": "56:f3:49:33:13:cf",
                                 "dns": {}
                             }]
                           k8s.v1.cni.cncf.io/networks: [{"interface":"net1","mac":"fe:35:dd:00:29:30","name":"public","namespace":"default"}]
                           k8s.v1.cni.cncf.io/networks-status:
                             [{
                                 "name": "",
                                 "interface": "eth0",
                                 "ips": [
                                     "10.52.0.230"
                                 ],
                                 "mac": "1a:7c:3c:8e:bc:5c",
                                 "default": true,
                                 "dns": {}
                             },{
                                 "name": "default/public",
                                 "interface": "net1",
                                 "mac": "56:f3:49:33:13:cf",
                                 "dns": {}
                             }]
                           kubevirt.io/domain: 79bec1b1-12c6-49e5-8e8e-b0ae4469e4af
Status:                    Terminating (lasts 33m)
Termination Grace Period:  30s
IP:                        10.52.0.230
IPs:
  IP:           10.52.0.230
Controlled By:  VirtualMachineInstance/79bec1b1-12c6-49e5-8e8e-b0ae4469e4af
Containers:
  compute:
    Container ID:  
    Image:         registry.suse.com/harvester-beta/virt-launcher:0.40.0-1
    Image ID:      
    Port:          <none>
    Host Port:     <none>
    Command:
      /usr/bin/virt-launcher
      --qemu-timeout
      5m
      --name
      79bec1b1-12c6-49e5-8e8e-b0ae4469e4af
      --uid
      f0da20ff-5cca-4c1e-a454-484447f1e4f7
      --namespace
      default
      --kubevirt-share-dir
      /var/run/kubevirt
      --ephemeral-disk-dir
      /var/run/kubevirt-ephemeral-disks
      --container-disk-dir
      /var/run/kubevirt/container-disks
      --grace-period-seconds
      45
      --hook-sidecars
      0
      --less-pvc-space-toleration
      10
      --ovmf-path
      /usr/share/OVMF
    State:          Waiting
      Reason:       ContainerCreating
    Last State:     Terminated
      Reason:       ContainerStatusUnknown
      Message:      The container could not be located when the pod was deleted.  The container used to be Running
      Exit Code:    137
      Started:      Mon, 01 Jan 0001 00:00:00 +0000
      Finished:     Mon, 01 Jan 0001 00:00:00 +0000
    Ready:          False
    Restart Count:  0
    Limits:
      devices.kubevirt.io/kvm:        1
      devices.kubevirt.io/tun:        1
      devices.kubevirt.io/vhost-net:  1
    Requests:
      cpu:                            100m
      devices.kubevirt.io/kvm:        1
      devices.kubevirt.io/tun:        1
      devices.kubevirt.io/vhost-net:  1
      ephemeral-storage:              50M
      memory:                         4490002433
    Environment:
      KUBEVIRT_RESOURCE_NAME_default:  
    Mounts:
      /var/run/kubevirt-ephemeral-disks from ephemeral-disks (rw)
      /var/run/kubevirt/container-disks from container-disks (rw)
      /var/run/kubevirt/hotplug-disks from hotplug-disks (rw)
      /var/run/kubevirt/sockets from sockets (rw)
      /var/run/libvirt from libvirt-runtime (rw)
    Devices:
      /dev/rootdisk from rootdisk
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  sockets:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  rootdisk:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  79bec1b1-12c6-49e5-8e8e-b0ae4469e4af-rootdisk-1z8mz
    ReadOnly:   false
  virt-bin-share-dir:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  libvirt-runtime:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  ephemeral-disks:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  container-disks:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  hotplug-disks:
    Type:        EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:      
    SizeLimit:   <unset>
QoS Class:       Burstable
Node-Selectors:  kubevirt.io/schedulable=true
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason   Age   From     Message
  ----    ------   ----  ----     -------
  Normal  Killing  33m   kubelet  Stopping container compute
```

**Comment by larssb** [comment]
I experienced the same thing. Though the entities that I saw this happening with was Pods I deployed, not through the Harvester UI, but KubeVirt VM's deployed with a tool called [kcli](https://github.com/karmab/kcli). Suddenly, one day, out of the blue, these kcli deployed KubeVirt VM's (running K3s) where not responding and they where running on the Harvester master node (the first one deployed) - I had to restart the Harvester master node before things started working again for these KubeVirt K3s VM's.

# SETUP

- Harvester v0.2.0
- 3 Harvester nodes
- Bare-metal deployment, via the Harvester ISO installation method
- Harvester is installed on SSD's

**Comment by gitlawr** [comment]
Can you provide a support bundle to help us investigate? You can generate that at the bottom of the UI

**Comment by inmanturbo** [comment]
Looking at these parameters
```yaml
      --ovmf-path
      /usr/share/OVMF
```
Is there ovmf support? I don't see any options for uefi bios in the UI

**Comment by Jellyfrog** [comment]
Not sure if it's the same, but I rebooted one host, and then the VM was dead and can't start:
[supportbundle_fb3938fa-4c03-4c20-984c-bc4a41394efc_2022-01-13T14-35-19Z.zip](https://github.com/harvester/harvester/files/7863570/supportbundle_fb3938fa-4c03-4c20-984c-bc4a41394efc_2022-01-13T14-35-19Z.zip)

**Comment by fanlix** [comment]
2 vm suddenly stuck at "stopping" state, cannot stop/start.

hv: v1.0.1
one node in use


```
Status:                    Terminating (lasts 17h)
Termination Grace Period:  30s
IP:                        10.52.0.167
IPs:
  IP:           10.52.0.167
Controlled By:  VirtualMachineInstance/fx
Containers:
  compute:
    Container ID:  containerd://07f05314c698c40df184af28c0a80c8955f29018f3d3b783880fa0f16d993178
    Image:         registry.suse.com/harvester-beta/virt-launcher:0.49.0-2
    Image ID:      sha256:0de0705b81ed0821151b35da38d4e505eac68ec70b2f1b3cd070a5ca016052a1
    Port:          <none>
    Host Port:     <none>
    Command:
      /usr/bin/virt-launcher
      --qemu-timeout
      356s
      --name
      .....
    State:          Terminated
      Reason:       Error
      Exit Code:    137
      Started:      Wed, 01 Jun 2022 03:43:47 +0000
      Finished:     Wed, 01 Jun 2022 10:22:28 +0000
    Ready:          False
    Restart Count:  0
```

i have to force delete the pods by hand:
```kubectl delete pod --force --grace-period=0  -n xxx virt-launcher-xxx-xxxxx```

**Comment by ibrokethecloud** [comment]
@fanlix are you please able to provide us a recent support bundle?

**Comment by fanlix** [comment]
@ibrokethecloud  sorry reply so late.   
Is there any way to upload the file not publicly?

**Comment by w13915984028** [comment]
@fanlix  Your VM container stopped was due to:
```
    State:          Terminated
      Reason:       Error
      Exit Code:    137
```

> Exit code 137 implies that a container within your Kubernetes pod was killed due to an Out Of Memory (OOM) error. This error occurs when the system runs out of memory and the Linux kernel kills processes to free up memory.

**Comment by w13915984028** [comment]
We assume the phenomenon `VM is stuck in terminating after a reboot` is same as #4033, it will be solved in Harvester v1.3.0.
